src/vm/lib/vm_string.c

```diff
@@ -6,27 +6,37 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdbool.h>
 
 int vm_strcmpl(const char *restrict s, const char *restrict s_) {
-    if (strlen(s) != strlen(s_)) {
+    int length_s = strlen(s);
+    int length_s_ = strlen(s_);
+
+    if (length_s != length_s_) {
         return 0;
     }
-
-    return vm_strncmpl(s, s_, strlen(s));
+    return vm_strncmpl(s, s_, length_s);
 }
 
 int vm_strncmpl(const char *restrict s, const char *restrict s_, int n) {
+    int length_s = strlen(s);
```

Comment:
The same reason goes here. The `vm_strncmpl` is quite simple and easy to understand, those `are_valid_indexes` is quite complex to read, don't you think? It's two lines long to say the same thing as the for describes.

There's no reason to abstract the length of the string variables as this function uses it only once in the original form. Revert those changes